### PR TITLE
Don't safeguard empty strings while reading

### PR DESCRIPTION
**Pull request by franzpoeschel**
This is a workaround/fix for #1221

### CODE REVIEW

File: include/openPMD/backend/Attributable.hpp [hunk 1]
```diff
@@ -90,21 +90,37 @@ namespace internal
         A_MAP m_attributes;
     };
 
+    enum class SetAttributeMode : char
+    {
+        WhileReadingAttributes,
+        FromPublicAPICall
```
Comment (ax3l) [inline include/openPMD/backend/Attributable.hpp:96]: Is that public API call both read and write? Might need a doxygen string, even though its readable already :)

Comment (ax3l) [reply include/openPMD/backend/Attributable.hpp:96]: Ah, `Set`... so write

File: include/openPMD/backend/Attributable.hpp [hunk 2]
```diff
@@ -410,11 +433,28 @@ OPENPMD_protected
...
+}
+
+inline bool
+Attributable::setAttribute(std::string const &key, char const value[])
+{
+    return setAttributeImpl(
+        key, value, internal::SetAttributeMode::FromPublicAPICall);
+}
+
+// TODO explicitly instantiate Attributable::setAttribute for all T in Datatype
```
Comment (ax3l) [inline include/openPMD/backend/Attributable.hpp]: This is done in `src/backend/Attributable.cpp`
```suggestion
// note: we explicitly instantiate Attributable::setAttributeImpl for all T in Datatype in Attributable.cpp
```

**Review by ax3l — APPROVED** [review]
Thank you, LGTM :)

**Comment by ax3l** [comment]
Will fix merge conflict from the merge of #1226

File: src/backend/Attributable.cpp [hunk 3]
```diff
@@ -400,23 +487,38 @@ void Attributable::readAttributes(ReadMode mode)
...
         case DT::VEC_CDOUBLE:
-            setAttribute(att, a.get<std::vector<std::complex<double> > >());
+            setAttributeImpl(
+                att,
+                a.get<std::vector<std::complex<double> > >(),
+                internal::SetAttributeMode::WhileReadingAttributes);
             break;
         case DT::VEC_CLONG_DOUBLE:
             setAttribute(
                 att, a.get<std::vector<std::complex<long double> > >());
```
Comment (ax3l) [inline src/backend/Attributable.cpp]: Is this one missing?
```suggestion
            setAttributeImpl(
                att,
                a.get<std::vector<std::complex<long double> > >(),
                internal::SetAttributeMode::WhileReadingAttributes);
```

**Review by ax3l — APPROVED** [review]

File: src/backend/Attributable.cpp (hunk 3 above)
Comment (ax3l) [reply src/backend/Attributable.cpp]: Added to last commit.

File: include/openPMD/backend/Attributable.hpp (hunk 2 above)
Comment (ax3l) [reply include/openPMD/backend/Attributable.hpp]: added to 2nd commit.

File: include/openPMD/backend/Attributable.hpp (hunk 1 above)
Comment (franzpoeschel) [reply include/openPMD/backend/Attributable.hpp:96]: It's write-only in the public API, but during parsing, we emulate read_write mode and use `setAttribute` also internally

File: src/backend/Attributable.cpp (hunk 3 above)
Comment (franzpoeschel) [reply src/backend/Attributable.cpp]: Looks like it, good catch

**Comment by ax3l** [comment]
I have trouble backporting this to the `release-0.14.5` branch due to changes in `include/openPMD/backend/Attributable.hpp`

**Comment by ax3l** [comment]
Applied to `0.14.5` via https://github.com/ax3l/openPMD-api/pull/6 :tada:

**Pull request merged**